wgpu-core/src/pipeline.rs

```diff
@@ -81,16 +79,16 @@ impl<B: hal::Backend> Borrow<RefCount> for ComputePipeline<B> {
 
 #[repr(C)]
 #[derive(Debug)]
-pub struct RenderPipelineDescriptor {
+pub struct RenderPipelineDescriptor<'a> {
     pub layout: PipelineLayoutId,
     pub vertex_stage: ProgrammableStageDescriptor,
-    pub fragment_stage: *const ProgrammableStageDescriptor,
+    pub fragment_stage: Option<&'a ProgrammableStageDescriptor>,
```

Comment:
I've used `Option` of references around here. I think this should be FFI-safe.

Reply:
Option of references works well when we are going to map the structure to C API.
This isn't the case here, so we should just use `Option<value>`, and also remove repr(C)